learning_journal/tests.py

```diff
@@ -1,30 +1,88 @@
 import pytest
 from pyramid import testing
+from .models import (
+        Entry,
+        get_engine,
+        get_session_factory,
+        get_tm_session
+)
+from .models.meta import Base
+import transaction
 
 
-def test_home_view():
-    from .views import home_view
-    request = testing.DummyRequest()
-    info = home_view(request)
-    assert info['title'] == 'Home'
+@pytest.fixture(scope="session")
+def sqlengine(request):
+    config = testing.setUp(settings={
+        'sqlalchemy.url': 'sqlite:///:memory:'
+    })
+    config.include(".models")
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
 
+    def teardown():
+        testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
 
-def test_single_entry_view():
-    from .views import single_entry
-    request = testing.DummyRequest()
-    info = single_entry(request)
-    assert info['title'] == 'Single entry'
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope="function")
+def new_session(sqlengine, request):
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+def test_entry_model(new_session):
+    assert len(new_session.query(Entry).all()) == 0
+    test_entry = Entry(title='Test entry', body='test body')
+    new_session.add(test_entry)
+    new_session.flush()
+    assert len(new_session.query(Entry).all()) == 1
+
+
+def dummy_http_request(new_session):
+    test_request = testing.DummyRequest()
+    test_request.dbsession = new_session
+    return test_request
+
+
+def test_home_view(new_session):
+    from .views.default import home_view
+    new_session.add(Entry(title='Pytest', body='<h1>This is a pytest</h1>'))
+    new_session.flush()
+    http_request = dummy_http_request(new_session)
+    result = home_view(http_request)
+    assert result['journal_entries'][-1].title == 'Pytest'
+
+
+def test_single_entry_view(new_session):
+    from .views.default import single_entry
+    new_session.add(Entry(title='Pytest', body='<h1>This is a pytest</h1>'))
+    new_session.flush()
+    http_request = dummy_http_request(new_session) 
+    http_request.matchdict['id'] = 1
+    result = single_entry(http_request)
+    assert result['journal'].title == 'Pytest'
 
 
 def test_new_entry_view():
-    from .views import new_entry
+    from .views.default import new_entry
     request = testing.DummyRequest()
     info = new_entry(request)
     assert info['title'] == 'New entry'
```

Comment:
Can you verify that the new entry has been added to the session after running the new_entry view?  Isn't that what should happen?